src/protocols/secure_channel/CASESession.cpp

```diff
@@ -1702,7 +1704,9 @@ CHIP_ERROR CASESession::HandleSigma3a(System::PacketBufferHandle && msg)
 
         SuccessOrExit(err = helper->ScheduleWork());
         mHandleSigma3Helper = helper;
-        mExchangeCtxt->WillSendMessage();
+
+        // When we receive sigma3, peer has received and processed sigma2 which means it is active
+        mExchangeCtxt->WillSendMessage(true /* IsPeerActive */);
```

Comment:
This code and new comment here clearly highlight that "is peer active" should be assumed true always during an exchange, and the value of "false" that would have been put here should be explicitly set in places where needed.

Reply:
resolving since the this changes doesn't exist anymore